src/node/node.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_NODE_H
+#define BITCOIN_NODE_NODE_H
+
+#include <list>
+#include <memory>
+
+class BanMan;
+class CConnman;
+class PeerLogicValidation;
+namespace interfaces {
+class Chain;
+class ChainClient;
+} // namespace interfaces
+using ChainClients = std::list<std::reference_wrapper<interfaces::ChainClient>>;
+
+//! Node struct containing chain state and connection state.
+//!
+//! More state could be moved into this struct to eliminate global variables,
+//! and allow creating multiple instances of validation and networking objects
```

Comment:
So you would create multiple Node in one test process, connect them and let them talk maybe with random introduction of interferences in-between ?